### ISSUE TITLE
Unable to use environment variables when publishing to npm

### ISSUE THREAD

**Issue by atrauzzi** [travis-ci/travis-ci#6941]
It would be nice if I could simply provide my key in the web UI and then use it in my npm deploy configuration.  Not sure if it's possible currently as I'm getting this error: https://travis-ci.org/atrauzzi/protoculture/builds/179159141

**Comment by BanzaiMan** [comment]
There isn't a predefined name which `npm` looks up when it tries to publish modules, is there? I don't see it in https://docs.npmjs.com/getting-started/publishing-npm-packages. If I missed them, do let me know.

You may use `NPM_EMAIL` and `NPM_TOKEN`, but others may use something different. Unless there is a convention understood by everyone, I don't think we can support such environment variables.

**Comment by BanzaiMan** [comment]
Oh. I see. You want to use them in your deployment configuration.

My bad. I misunderstood. I'll think about this a little more.

**Comment by atrauzzi** [comment]
Yeah, right now the documentation would have me putting secrets in my project, which I'm not about to do.  Even if they are encrypted.

All I should expect to do beyond configuring Travis to use npm deployments is get my own key and then put it in the web interface. Nowhere else.

**Comment by atrauzzi** [comment]
@BanzaiMan - Any further thoughts/status on this one?

**Comment by BanzaiMan** [comment]
Sorry. I haven't had time to look into this further.

**Comment by BanzaiMan** [comment]
I tweaked the code to dump what is getting executed there. https://travis-ci.org/BanzaiMan/travis_production_test/builds/196025957#L232

With [this configuration](https://github.com/BanzaiMan/travis_production_test/blob/952eeeab482be8d8fc6fd7614f080637701ba7a2/.travis.yml), we set the values correctly and execute `env NPM_API_KEY=deadbeef npm publish` (with credentials written to `~/.npmrc`). This suggests to me that the command invocation is correct. If you are getting the error, then I have to think that your `NPM_KEY` might be somehow faulty.
